### PR TITLE
feat: support added scrape_protocols to Pod/Service monitors

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

support added scrape_protocols to Pod/Service monitors

ref: https://github.com/prometheus-operator/prometheus-operator/pull/6235#pullrequestreview-1830604813

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
support added scrape_protocols to Pod/Service monitors
```

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
I don't think that the operator should take into account the global config. If a service/pod monitor defines the list of protocols, it should be applied as-is.

**Comment by dongjiang1989** [comment]
> I don't think that the operator should take into account the global config. If a service/pod monitor defines the list of protocols, it should be applied as-is.

Thanks. I got it.

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -259,6 +259,17 @@ func (cg *ConfigGenerator) AddTrackTimestampsStaleness(cfg yaml.MapSlice, trackT
 	return cg.WithMinimumVersion("2.48.0").AppendMapItem(cfg, "track_timestamps_staleness", *trackTimestampsStaleness)
 }
 
+// AddScrapeProtocols adds the scrape_protocols field into scrape configurations.
+// When the global scrapeProtocols not empty, override applies.
```
Comment (dongjiang1989) [inline pkg/prometheus/promcfg.go:263]:
```suggestion
```

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**